src/uci.cpp

```diff
@@ -67,6 +68,10 @@ namespace {
     else
         return;
 
+    string rfen = fen;
+    regex c("\\s*([rnbqkpRNBQKP1-8]+\\/){7}([rnbqkpRNBQKP1-8]+)\\s[bw-]\\s(([a-hkqA-HKQ]{1,4})|(-))\\s(([a-h][3-6])|(-))\\s\\d+\\s\\d+ \\s*");
+    if (regex_match(rfen, c));  else { cout << "Invaild fen" << endl; return; }
```

Comment:
```c++
if (!regex_match(rfen, c)) {
    cout << "Invalid fen" << endl;
    return;
}
```

Reply:
Moreover, makes a ton of sense here to look into C++ raw string literals (6. of [this link](https://en.cppreference.com/w/cpp/language/string_literal)) to limit the number of escape backslashes.

Reply:
Thanks, but the problem is not solved after changing the "if...else statement"